### PR TITLE
Tickets/dm 41338

### PR DESCRIPTION
**Pull request by dsanmartim**
I am opening a draft PR so you can take a look at the code. 

I have only implemented the "slew_to_planet" functionality in the "base_track_target.py" script, along with a few unit tests in both "test_auxtel_track_target.py" and "test_maintel_track_target.py." The new test cases for the new functionality are functioning correctly, but some of the old ones are now failing. I believe these errors are connected to the schema in "base_track_target.py," but despite trying numerous approaches, none have been successful so far.

### CODE REVIEW

File: python/lsst/ts/standardscripts/base_track_target.py [hunk 1]
```diff
@@ -278,14 +299,20 @@ async def configure(self, config):
...
 
         self.log.debug(f"Slew type: {self.slew_type!r}.")
 
         await self.configure_tcs()
 
-        self.config.rot_type = getattr(RotType, self.config.rot_type)
+        if self.slew_type is not SlewType.PLANET:
+            self.config.rot_type = getattr(RotType, self.config.rot_type)
+        else:
+            self.config.rot_type = RotType.Sky
```
Comment (tribeiro) [inline python/lsst/ts/standardscripts/base_track_target.py:314]: I think this has no effect if the slew type is `PLANET`. I think we have 2 option here either raise a `RuntimeError` exception if the user select `PLANET` with `rot_type != Sky` or log a warning message. I am really torn between both options, but I slightly lean towards just warning and giving it a pass..

File: python/lsst/ts/standardscripts/base_track_target.py [hunk 2]
```diff
@@ -347,6 +374,22 @@ async def run_block(self):
...
                 slew_timeout=self.slew_timeout,
             )
+
+        elif self.slew_type == SlewType.PLANET:
+            planet_name = self.config.slew_planet["planet_name"]
+            planet_enum = Planets[planet_name]
+
+            self.log.info(
+                f"Slew and track planet_name={planet_name} rot_value={self.config.rot_value} "
+                f"rot_type={self.config.rot_type} "
```
Comment (tribeiro) [inline python/lsst/ts/standardscripts/base_track_target.py:384]: Since this is always `rot_sky`, replace this with `rot_sky={...}` and remove the `rot_type`.

File: tests/test_auxtel_track_target.py [hunk 3]
```diff
@@ -144,6 +172,16 @@ def assert_slew_azel(self, find_target_config):
         self.script.tcs.find_target.assert_awaited_with(**find_target_config)
         self.assert_slew_target_name()
 
+    def assert_slew_planet(self, planet_name):
+        planet_enum = Planets[planet_name]
+        self.script.tcs.slew_to_planet.assert_awaited_once()
+        self.script.tcs.slew_to_planet.assert_awaited_with(
+            planet=planet_enum,
+            rot_sky=self.script.config.slew_planet.get("rot_sky", 0.0),
```
Comment (tribeiro) [inline tests/test_auxtel_track_target.py:178]: I think this needs updating

File: tests/test_auxtel_track_target.py [hunk 4]
```diff
@@ -46,6 +47,16 @@ async def basic_make_script(self, index):
...
         return (self.script,)
 
+    async def test_configure_with_slew_planet(self):
+        async with self.make_script():
+            self.script._atcs = unittest.mock.AsyncMock()
+            self.script._atcs.start_task = utils.make_done_future()
+
+            # Script can be configured with slew_planet
+            await self.configure_script(
+                slew_planet=dict(planet_name="PLUTO", rot_sky=0.0)
```
Comment (tribeiro) [inline tests/test_auxtel_track_target.py]: You removed `rot_sky` right?

File: tests/test_auxtel_track_target.py [hunk 5]
```diff
@@ -98,6 +109,23 @@ async def test_run_slew_azel(self):
...
+    async def test_run_slew_planet(self):
+        async with self.make_script():
+            self.script.tcs.slew_to_planet = unittest.mock.AsyncMock()
+            self.script.tcs.stop_tracking = unittest.mock.AsyncMock()
+
+            # Get planet name
+            planet_name = Planets["PLUTO"].name
+
+            # Check running with planet name
+            config = dict(slew_planet=dict(planet_name=planet_name, rot_sky=0.0))
```
Comment (tribeiro) [inline tests/test_auxtel_track_target.py]: You removed `rot_sky` right?

File: tests/test_maintel_track_target.py [hunk 6]
```diff
@@ -144,6 +144,16 @@ async def test_configure_with_target_name_program_reason(self):
...
             )
 
+    async def test_configure_with_slew_planet(self):
+        async with self.make_script():
+            self.script._mtcs = unittest.mock.AsyncMock()
+            self.script._mtcs.start_task = utils.make_done_future()
+
+            # Script can be configured with slew_planet
+            await self.configure_script(
+                slew_planet=dict(planet_name="PLUTO", rot_sky=0.0)
```
Comment (tribeiro) [inline tests/test_maintel_track_target.py]: remove rot_sky.

**Review by tribeiro — COMMENTED** [review]
Here some comments for you to consider.

File: tests/test_auxtel_track_target.py (hunk 4 above)
Comment (dsanmartim) [reply tests/test_auxtel_track_target.py]: I realized that while running the unit test. All unit tests passing now.

File: python/lsst/ts/standardscripts/base_track_target.py [hunk 7]
```diff
@@ -278,14 +299,22 @@ async def configure(self, config):
...
 
         await self.configure_tcs()
 
-        self.config.rot_type = getattr(RotType, self.config.rot_type)
+        if self.slew_type == SlewType.PLANET and self.config.rot_type != "Sky":
+            self.log.warning(
+                f"Rotator type must be Sky for slew_type={self.slew_type}. "
+            )
+        else:
+            self.config.rot_type = getattr(RotType, self.config.rot_type)
```
Comment (tribeiro) [inline python/lsst/ts/standardscripts/base_track_target.py:316]: I think it would be better if you did this before the `if` statement, then, compare `self.config.rot_type` with `RotType.Sky`. Also, since you are not raising an exception, but issuing a warning message, maybe rephrase the message a bit.

Consider the following:

```
self.config.rot_type = getattr(RotType, self.config.rot_type)

if self.slew_type == SlewType.PLANET and self.config.rot_type != RotType.Sky:
    self.log.warning(
        f"Slew_type={self.slew_type!r} only works for {RotType.Sky!r}, "
        f"got {self.config.rot_type!r}. Ignoring."
    )
```

File: python/lsst/ts/standardscripts/base_track_target.py [hunk 8]
```diff
@@ -347,6 +376,21 @@ async def run_block(self):
                 time_on_target=self.config.track_for,
                 slew_timeout=self.slew_timeout,
             )
+
+        elif self.slew_type == SlewType.PLANET:
+            planet_name = self.config.slew_planet["planet_name"]
+            planet_enum = Planets[planet_name]
+
+            self.log.info(
+                f"Slew and track planet_name={planet_name} rot_sky={self.config.rot_value} "
```
Comment (tribeiro) [inline python/lsst/ts/standardscripts/base_track_target.py:385]: There's an extra space at the end. Maybe add a "."?

**Review by tribeiro — APPROVED** [review]
Looks good, but please, take a look at my comments before merging.
